crates/cranelift/src/compiler/component.rs

```diff
@@ -1331,6 +1336,26 @@ impl<'a> TrampolineCompiler<'a> {
         );
     }
 
+    fn translate_thread_spawn_indirect(&mut self, table: RuntimeTableIndex) {
+        let args = self.builder.func.dfg.block_params(self.block0).to_vec();
+        let vmctx = args[0];
+        let element = args[1];
+        let context = args[2];
+
+        // table: u32
+        let table = self
+            .builder
+            .ins()
+            .iconst(ir::types::I32, i64::from(self.offsets.runtime_table(table)));
```

Comment:
This is a disconnect between runtime and compile-time where here the u32 is the offset in the VMContext but at runtime it's interpreted as `RuntimeTableIndex`